src/backend/discovery/discovery.lambda.ts

```diff
@@ -0,0 +1,291 @@
+      }).catch(function (error) {
+        throw Error(`Error while processing batch, marker will not be updated, exiting.\n ${error}`);
+      });
+    })
+    .on('end', () => {
+      console.log('Changes feed monitoring has stopped');
+    });
+}
+
+async function writeMarkerToS3(sequence: Number) {
+  const client = new AWS.S3();
+  await client.putObject({
+    Bucket: stagingBucket,
+    Key: MARKER_FILE_NAME,
+    Body: sequence.toString(),
+    Metadata: {
+      'Lambda-Log-Group': functionContext.logGroupName,
+      'Lambda-Log-Stream': functionContext.logStreamName,
+      'Lambda-Run-Id': functionContext.awsRequestId,
+    },
+  }).promise();
+}
+
+async function processPackageUpdate(change: Change) {
+  let latestVersion: VersionInfo;
+  try {
+    // we assume that the change is concerning the latest version
+    latestVersion = getLatestVersion(change);
+    if (!isJsiiModule(latestVersion) || !isConstruct(latestVersion)) {
+      return;
+    }
+  } catch (error) {
+    console.error(`Could not parse change for ${change.id}`);
+    throw error;
+  }
+
+  return new Promise<void>((ok, ko) => {
+    // change.dist.tarball => https://registry.npmjs.org/<@scope>/<name>/-/<name>-<version>.tgz
+    // staging bucket key  => packages/<@scope>/<name>/-/<name>-<version>.tgz
+    const key = latestVersion.dist.tarball.replace('https://registry.npmjs.org', 'packages');
+    copyPackageToS3(latestVersion, key).then(() => {
+      return sendSqsMessage(latestVersion, key);
+    }).catch(function(reason) {
+      console.error(`Error processing change for ${change.id}:\n ${reason}`);
+      ko();
+    });
+  });
+}
+
+/**
+ * Copy the tarball from the npm registry to S3
+*/
+async function copyPackageToS3(versionInfo: VersionInfo, key: string): Promise<void> {
+  console.log(`uploading tarball to s3, key: ${key}`);
+  return new Promise<void>((ok, ko) => {
+    https.get(versionInfo.dist.tarball, function (response /* Readable */) {
+      s3.putObject({
+        Bucket: stagingBucket,
+        Key: key,
+        Body: response,
+        Metadata: {
+          'Lambda-Log-Group': functionContext.logGroupName,
+          'Lambda-Log-Stream': functionContext.logStreamName,
+          'Lambda-Run-Id': functionContext.awsRequestId,
+        },
+      }).promise().then(() => ok(), ko);
+    }).on('error', function (error) {
+      console.error(`Error attempting to stage tarball in S3: ${error}`);
+      ko(`Failed downloading file from ${versionInfo.dist.tarball}`);
+    });
+  });
+}
+
+/**
+ * Send an SQS message to notify that a new version was published
+ * @param versionInfo the version info
+ * @param key the s3 key to which the tarball was uploaded
+ */
+async function sendSqsMessage(versionInfo: VersionInfo, key: string): Promise<void> {
+  console.log(`posting sqs message for ${versionInfo.packageName}`);
+  const message: Message = {
+    packageName: versionInfo.name,
+    time: new Date(versionInfo.time.modified),
```

Comment:
You actually don't need to parse the `modified` time here... It's already in the expected format.

```suggestion
    time: versionInfo.time.modified,
```